### PR TITLE
Upgrade dependancies and fix some stuff.

### PR DESCRIPTION
**Pull request by mstratford**
In theory this fixes Raygun redux reporting, was well as reducing errors and warnings all over the place.

I've had a play and the only thing I've seen is a sock-js error which i'm still tracing.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
+  persistStore,
+  FLUSH,
+  REHYDRATE,
+  PAUSE,
+  PERSIST,
+  PURGE,
+  REGISTER
+} from "redux-persist";
+
+const raygunMiddleware: Middleware < {}, RootState, Dispatch < any >> = (store) => (next) => (action) => {
```
Comment (markspolakovs) [inline src/store.ts:20]: I'm... not actually sure how this fixes anything? Looks identical to the old code, modulo type declarations

Comment (mstratford) [reply src/store.ts:20]: Me neither, but it stopped breaking it / complaining haha. I'm now seeing raygun.js appearing in the console again too.

Comment (markspolakovs) [reply src/store.ts]: Have you checked if it properly reports redux actions?

**Comment by mstratford** [comment]
Replaced by #152

**Pull request closed without merging**